main/src/hadd.cxx

```diff
@@ -606,6 +674,10 @@ int main(int argc, char **argv)
    if (args.fCacheSize)
       Info() << "Using " << cacheSize << "\n";
 
+   ////////////////////////////// end flags processing /////////////////////////////////
+
+   gSystem->Load("libTreePlayer");
```

Comment:
Why do we need this library?

Reply:
Not sure, it was there already (I just moved it past the flags processing in this PR). Perhaps @pcanal knows it.

Reply:
The dependency was originally a explicit link added to "allow support for TTree with a build index" and then was moved to a run-time dependency because the usage is hidden being a layer of indirect and was dropped in some case by the remove unused dependency feature of the linker (or something like that).

The original commit ended with "until hadd can use the autoloader" which is likely the case now.

So it probably can be technically removed.

 A simple test would to create  a TTree, Create and index, save it to the file, merge the result a couple of times and check the results.

Reply:
I guess we want to do this in a separate PR